hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/PipelineSyncTask.java

```diff
@@ -66,21 +71,34 @@ public PipelineSyncTask(ReconPipelineManager pipelineManager,
   @Override
   protected synchronized void run() {
     try {
+      lock.writeLock().lock();
       while (canRun()) {
-        long start = Time.monotonicNow();
-        List<Pipeline> pipelinesFromScm = scmClient.getPipelines();
-        reconPipelineManager.initializePipelines(pipelinesFromScm);
-        syncOperationalStateOnDeadNodes();
-        LOG.info("Pipeline sync Thread took {} milliseconds.",
-            Time.monotonicNow() - start);
-        recordSingleRunCompletion();
+        triggerPipelineSyncTask();
         wait(interval);
       }
     } catch (Throwable t) {
       LOG.error("Exception in Pipeline sync Thread.", t);
       if (t instanceof InterruptedException) {
         Thread.currentThread().interrupt();
       }
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public void triggerPipelineSyncTask()
+      throws IOException, TimeoutException, NodeNotFoundException {
+    try {
+      lock.writeLock().lock();
+      long start = Time.monotonicNow();
+      List<Pipeline> pipelinesFromScm = scmClient.getPipelines();
+      reconPipelineManager.initializePipelines(pipelinesFromScm);
+      syncOperationalStateOnDeadNodes();
```

Comment:
The trigger is on stale node, please check if applicable or not, as method indicate for deadnode sync.

Additionally do this required for Deadnode handler or healthCheck trigger is enough ?

Reply:
Thanks @sumitagrawl for review, its fixed. pls re-review.